### PR TITLE
Import Configuration Improvements

### PR DESCRIPTION
**Pull request by Nebby1999**
This PR fixes some issues the Import Configuration system had, alongside adding some new functionalities.

Fixes:
1.- Fixes a major bug regarding the ending process of the system, where the Cleanup procedure would not execute due to an error in the If statement before it

Improvements:
1.- Import Configuration now tries to update itself whenever the editor compiles via an InitializeOnLoad method, this in turn makes it possible to download an Importer extension and immediately see them on the ImportConfiguration without the need to delete the asset. This works by caching the current amount of optional executors in the app domain and then updating the asset if the count is different from the cached one.
2.- A lot of the code on Initialize has been moved to different methods in the ImportConfiguration class, this is due to the first improvement reusing a lot of the code that was found there
3.- Added a static utility method on ComposableObject that allows you to remove null sub assets while keeping the meta file intact
4.- Added a new Toggle on the ThunderKitSettings, this toggle is used to determine wether or not to notify the end user that the import process is complete using the EditorApplication.Beep() method.
5.- Most importantly, i've added a Restart prompt at the end of the import configuration, i've noticed that just after importing some unusual bugs occur (such as creating any custom monobehaviour cant be added to game objects), but these bugs disappear as soon as the editor is restarted.

I would've liked to make it so when the import process notifies the end user with the beep, to also force focus of the unity editor application to make sure they see the restart prompt, i couldnt find a way to do this however.

### CODE REVIEW

**Comment by PassivePicasso** [comment]
The lack of automatic focus is a good thing.  Id have rejected that feature anyway. 
Stopping Focus Theft starts at the developers desk.

**Comment by PassivePicasso** [comment]
Please add attribution for the missing script cleanup to the Third-party attribution markdown page.

File: Editor/Core/Config/ImportConfiguration.cs [hunk 1]
```diff
@@ -221,10 +265,23 @@ public void ImportGame()
...
+                PromptRestart();
+            }
+        }
+
+        private void PromptRestart()
+        {
+            if(GetOrCreateSettings<ThunderKitSettings>().notifyWhenImportCompletes)
+                EditorApplication.Beep();
+            
+            if(EditorUtility.DisplayDialog("Import Process Complete", "The game has been imported succesfully, It is extremely recommended to Restart your project to ensure stability", "Restart Project", "Restartt Later"))
```
[the remark this answers is not in the document]
Comment (PassivePicasso) [reply Editor/Core/Config/ImportConfiguration.cs:282]: "Restartt Later"

[the remark this answers is not in the document]
Comment (PassivePicasso) [reply Editor/Core/Config/ImportConfiguration.cs:282]: Some minor grammatical errors here, just update the message to:
"The game has been imported successfully. It is recommended to restart your project to ensure stability"

File: Editor/Core/Config/ImportConfiguration.cs [hunk 2]
```diff
@@ -221,10 +265,23 @@ public void ImportGame()
                 Debug.LogError($"Error during Import: {e}");
                 return;
             }
-            if (ConfigurationIndex > ConfigurationExecutors.Length)
+            if (ConfigurationIndex > ConfigurationExecutors.Length - 1)
```
Comment (PassivePicasso) [inline Editor/Core/Config/ImportConfiguration.cs:268]: minor nitpick,  I'd rather use >= here than -1
They are functionality identical here, this is merely preference

**Comment by PassivePicasso** [comment]
Initial install of ThunderKit in a fresh project results in duplicated ImportExtensions

image

**Comment by Nebby1999** [comment]
> Initial install of ThunderKit in a fresh project results in duplicated ImportExtensions
> 
> image

This has been fixed on commit 1cc82fc

**Pull request merged**